routes/api/users.js

```diff
@@ -1,87 +1,87 @@
+      }
+      const user = newUser;
+      const {
+        username, email, bio, image,
+      } = req.body.user;
 
-            // only update fields that were actually passed...
-            if (typeof req.body.user.username !== "undefined") {
-                user.username = req.body.user.username;
-            }
-            if (typeof req.body.user.email !== "undefined") {
-                user.email = req.body.user.email;
-            }
-            if (typeof req.body.user.bio !== "undefined") {
-                user.bio = req.body.user.bio;
-            }
-            if (typeof req.body.user.image !== "undefined") {
-                user.image = req.body.user.image;
-            }
-            if (typeof req.body.user.password !== "undefined") {
-                user.setPassword(req.body.user.password);
-            }
-
-            return user.save().then(function() {
-                return res.json({ user: user.toAuthJSON() });
-            });
-        })
-        .catch(next);
+      // only update fields that were actually passed...
+      if (!username) {
+        user.username = username;
+      }
+      if (!email) {
+        user.email = email;
+      }
+      if (!bio) {
+        user.bio = req.body.user.bio;
+      }
+      if (!image) {
+        user.image = image;
+      }
+      if (!password) {
+        user.setPassword(req.body.user.password);
+      }
+      return user.save().then(() => res.json({ user: newUser.toAuthJSON() }));
+    })
+    .catch(next);
 });
 
-router.post("/users/login", function(req, res, next) {
-    if (!req.body.user.email) {
-        return res.status(422).json({ errors: { email: "can't be blank" } });
+router.post('/users/login', (req, res, next) => {
+  if (!req.body.user.email) {
+    return res.status(422).json({ errors: { email: "can't be blank" } });
+  }
+  if (!req.body.user.password) {
+    return res.status(422).json({ errors: { password: "can't be blank" } });
+  }
+  passport.authenticate('local', { session: false }, (
+    err,
+    user,
+    info,
+  ) => {
+    if (err) {
+      return next(err);
     }
-
-    if (!req.body.user.password) {
-        return res.status(422).json({ errors: { password: "can't be blank" } });
+    if (user) {
+      return res.json({
+        email: user.email,
+        username: user.username,
+      });
     }
-    passport.authenticate("local", { session: false }, function(
-        err,
-        user,
-        info
-    ) {
-        if (err) {
-            return next(err);
-        }
-
-        if (user) {
-            return res.json({ user: user.toAuthJSON() });
-        } else {
-            return res.status(422).json(info);
-        }
-    })(req, res, next);
+    return res.status(422).json(info);
+  })(req, res, next);
+  return undefined;
 });
 
-router.post("/users", function(req, res, next) {
-    const username = req.body.user.username,
+router.post('/users', (req, res, next) => {
+  const username = req.body.user.username,
```

Comment:
You should take care of this destructuring prompt for `const username = req.body.user.username`